### DISCUSSION TITLE
Keine Verbindung zur Fritzbox über WLAN möglich

### DISCUSSION THREAD

**Discussion by Gloomyeye** [tbnobody/OpenDTU#576] [Q&A] [answered]
Hallo zusammen,
ich versuche verzweifelt OpenDTU mit meinem WLAN zu verbinden. SSID und Passwort sind korrekt eingegeben und DHCP ist aktiviert. Auf meiner Fritzbox habe ich gesagt, dass sich die WLAN Geräte untereinander unterhalten können (wie hier gesagt: https://avm.de/service/wissensdatenbank/dok/FRITZ-Box-7590/549_Datei-und-Druckerfreigaben-eines-Computers-sind-im-Heimnetz-nicht-sichtbar/ ). Ich habe auch mein Gastnetzwerk aktiviert, wobei ich das Passwort (Test1234) ganz einfach gewählt habe und auch dieses funktioniert nicht.
Anschließend habe ich eine feste IP-Adresse vergeben (auch in der Fritzbox so konfiguriert) und auch das ist nicht von Erfolg gekrönt.

Mein Log sieht so aus:
08:34:36.517 > WiFi disconnected
08:34:36.517 > Try reconnecting
08:34:36.580 > Network lost connection
08:34:37.413 > WiFi disconnected
08:34:37.413 > Try reconnecting
08:34:37.475 > Network lost connection
08:34:38.440 > WiFi disconnected
08:34:38.440 > Try reconnecting
08:34:38.503 > Network lost connection
08:34:39.462 > WiFi disconnected
08:34:39.462 > Try reconnecting
08:34:39.521 > Network lost connection
08:34:40.487 > WiFi disconnected
08:34:40.487 > Try reconnecting
08:34:40.551 > Network lost connection

In den Netzwerkeinstellungen sieht man leider keine IP-Adresse (Beispiel mit Gast-WLAN und DHCP aktiv): 
<img width="1006" alt="Screenshot 2023-02-09 084049" src="https://user-images.githubusercontent.com/69754128/217749132-d8f124c2-9063-45a7-b8f1-fb0f0c1df699.png">

Zwischenzeitlich habe ich Ahoy ausprobiert und hier hat die WLAN-Verbindung sofort geklappt. Verkabelung oder ESP32 schließe ich somit aus. Es muss daher an OpenDTU liegen. An meiner Netzwerkumgebung habe ich keine Änderungen vorgenommen.

Hat einer ein ähnliches Problem oder könnte mir helfen?

**Ausgangsituation**

- ESP-WROOM-32 + NRF24L01+
- OpenDTU Version "b8ced54". (aus https://github.com/tbnobody/OpenDTU/actions/runs/4047502490)
- Verbindungsdaten für WLAN korrekt hinterlegt
- FritzBox 7590 WLAN Router (v.7.50) + Mesh (mit FritzBox Repeater)
- PiHole ist als DNS aktiv
- 2,4 GHz + 5 GHz sind aktiviert (WP2+WPA3 Verschlüsselung aktiv)
- Abstand von ESP32 zum Router minimal (< 3m)

**Comment by Juergen2453** [comment]
Was mir auffällt ist das bei "WiFI Informationen (Access Point) -> # Teilnehmer 1"
Ist irgend ein Gerät noch mit dem Access Point verbunden? Und welche Version ist installiert?

**Reply by Gloomyeye** [reply]
Ich habe nun noch ein paar Punkte zur Ausgangssituation in den ersten Post mit aufgenommen.
Das sind zurzeit meine Einstellungen, wenn ich mich über den AP direkt auf dem ESP verbinde:
<img width="910" alt="Screenshot 2023-02-09 092903" src="https://user-images.githubusercontent.com/69754128/217758794-21770754-7d9b-47f0-adf4-46545ef671c7.png">
<img width="919" alt="Screenshot 2023-02-09 092932" src="https://user-images.githubusercontent.com/69754128/217758819-d9b2411c-6a05-4cdf-9bf8-f4dff1c0fe37.png">

<img width="908" alt="Screenshot 2023-02-09 092956" src="https://user-images.githubusercontent.com/69754128/217758838-c096a581-70d4-4891-9eeb-be53063d3366.png">

Die Informationen unter Netzqwerk suggerieren mir, dass ich einen IP-Adresse (bei einer fest eingestellten IP-Adresse in den Netzwerkeinstellungen) habe. Allerdings sieht man in der Konsole, dass er immer keine Verbindung zum Netzwerk hat bzw. immer wieder verliert.

**Comment by Juergen2453** [comment]
Warum machst du kein DHCP "Netzwerkeinstellungen -> DHCP aktivieren" ?

**Reply by Gloomyeye** [reply]
Auch diese Einstellung habe ich versucht. Ohne Erfolg.

**Reply by Gloomyeye** [reply]
das sind meine DHCP Einstellungen:

![17A2746B-5688-4ECB-A79D-5037D4DF8DD5](https://user-images.githubusercontent.com/69754128/217768526-5b9adc79-7c78-4bcc-96b7-b5d3c55d48cc.png)

**Comment by Gloomyeye** [comment, marked as answer]
Ich erhalte nun doch Zugriff auf mein Netzwerk, wenn ich einen festen Kanal in den Funknetzeinstellungen der Fritzbox hinterlege. Ich hatte vorher "Auto" als Einstellung gewählt. Bei Google habe ich gesehen, dass wohl die ESP-WROOM-32 Reihe Probleme bei den Funkkanälen von der Fritzbox hat.

**Comment by skippgh** [comment]
Der ESP sucht normalerweise nur nach dem ersten Kanal, der die gewünschte SSID enthält (siehe auch https://github.com/espressif/arduino-esp32/commit/3253de87 ). Ein erster schneller Patch in NetworkSettings.cpp zeigt auch, daß das Problem behoben werden kann:
`void NetworkSettingsClass::applyConfig()
{
    setHostname();
    if (!strcmp(Configuration.get().WiFi.Ssid, "")) {
        return;
    }
    MessageOutput.print("Configuring WiFi STA using ");
    if (strcmp(WiFi.SSID().c_str(), Configuration.get().WiFi.Ssid) || strcmp(WiFi.psk().c_str(), Configuration.get().WiFi.Password)) {
        MessageOutput.print("new credentials... ");
        WiFi.begin(
            Configuration.get().WiFi.Ssid,
            Configuration.get().WiFi.Password,
            WIFI_ALL_CHANNEL_SCAN);  // <- added here  '
Schöner wäre es natürlich, das über einen parametrisierbaren Config_switch zu lösen, dafür würde ich aber noch etwas Einarbeitungszeit in das Projekt benötigen. Jemand da, der das schneller könnte.

**Reply by CosmicDuck** [reply]
Sinnvoller wäre einen Signalvergleich zu machen und mit dem verbinden der die größte Signalstärke hat.

**Reply by CosmicDuck** [reply]
Achso, und wenn das nicht funktioniert den nächsten zu versuchen.

**Reply by skippgh** [reply]
Ja genau an der Stelle muss man auch noch patchen in:
`void NetworkSettingsClass::NetworkEvent(const WiFiEvent_t event)
{.....
case ARDUINO_EVENT_WIFI_STA_DISCONNECTED:

        MessageOutput.println("WiFi disconnected");
        if (_networkMode == network_mode::WiFi) {
            MessageOutput.println("Try reconnecting");
            // WiFi.reconnect();
            WiFi.disconnect();
            WiFi.begin(
                Configuration.get().WiFi.Ssid,
                Configuration.get().WiFi.Password,
                WIFI_ALL_CHANNEL_SCAN);
            raiseEvent(network_event::NETWORK_DISCONNECTED);
        }`
Also anstatt des WiFI.reconnect() nochmal ein expliziten neuen Scan starten. Das läuft im Moment bei mir zumindest stabil ....

**Reply by skippgh** [reply]
Ach ja und das mit dem Channel Scan und der Auswahl des signalstärksten Kanals macht der ESP mit seiner Lib auch selber in:
`
void NetworkSettingsClass::init(Scheduler& scheduler)
{
    using std::placeholders::_1;

    WiFi.setScanMethod(WIFI_ALL_CHANNEL_SCAN);
    WiFi.setSortMethod(WIFI_CONNECT_AP_BY_SIGNAL);
`

**Comment by tbnobody** [comment]
Könnt ihr mal folgende Änderung versuchen? (Und zwar bitte nur diese, ohne eure Patches im Event Handler etc)

```diff
diff --git a/include/defaults.h b/include/defaults.h
index e783df2..0070e09 100644
--- a/include/defaults.h
+++ b/include/defaults.h
@@ -13,7 +13,7 @@
 #define AUTH_USERNAME "admin"
 #define SECURITY_ALLOW_READONLY true

-#define WIFI_RECONNECT_TIMEOUT 15
+#define WIFI_RECONNECT_TIMEOUT 30
 #define WIFI_RECONNECT_REDO_TIMEOUT 600

 #define WIFI_SSID ""
diff --git a/src/NetworkSettings.cpp b/src/NetworkSettings.cpp
index 92a3156..c8004f1 100644
--- a/src/NetworkSettings.cpp
+++ b/src/NetworkSettings.cpp
@@ -268,7 +268,8 @@ void NetworkSettingsClass::applyConfig()
         MessageOutput.print("new credentials... ");
         WiFi.begin(
             Configuration.get().WiFi.Ssid,
-            Configuration.get().WiFi.Password);
+            Configuration.get().WiFi.Password,
+            WIFI_ALL_CHANNEL_SCAN);
     } else {
         MessageOutput.print("existing credentials... ");
         WiFi.begin();
```

**Reply by skippgh** [reply]
Ist bisher (4h) mit diesem Setup ohne Probleme gelaufen.

**Reply by skippgh** [reply]
auch nach 1.5 Tagen noch stabile Verbindung. Kommt der Patch dann in den offiziellen Code ?

**Reply by tbnobody** [reply]
Jeb, auf jeden Fall.

**Reply by djiwondee** [reply]
Hallo @tbnobody ,
hat denn dieser patch in der Zwischenzeit Einzug gehalten?

Ich habe seit Kurzem in der Tat Probleme, dass der ESP32 die Wifi-Connection verliert. Ja, ich habe mein Wifi Setup mit einer fritzbox 7590 und einem Repeater R3000 verändert. Die beiden Geräte bilden ein Mesh mittels LAN-Brücke (nicht **W**LAN-Brücke)

Ich habe aufgrund der Fehler mal eben heute ein "frisches" ESP32-NodeMCU mit dem mit einem [Webinstaller](https://www.opendtu.solar/firmware/webinstall/) geflashed. Eine Wifi-Verbindung kommt nur sporadisch zustande

```
Starting OpenDTU
Initialize FS... done
Reading configuration... done
Reading PinMapping... found valid mapping done
Initialize Network... done
Setting Hostname... Configuring WiFi STA using new credentials... done
Initialize NTP... done
Initialize SunPosition... done
Initialize MqTT... done
Initialize WebApi... done
Initialize Display... done
Initialize LEDs... done
Check for default DTU serial... done
Initialize Hoymiles interface... NRF: Connection successful
  Setting radio PA level... 
  Setting DTU serial... 
  Setting poll interval... 
  Adding inverter: {{myInverter Serial and Name}}  done
done
Switch to WiFi mode
Setting Hostname... done
Configuring WiFi STA using new credentials... done
Configuring WiFi STA DHCP IP... done
WiFi connected
Fetch inverter: {{myInverter Serial}}
WiFi got ip: 192.168.178.28
Network connected
Connecting to MQTT...
Connected to MQTT.
Fetch inverter: {{myInverter Serial}}
Admin AP remaining seconds: 10 / 600
Fetch inverter: {{myInverter Serial}}
Admin AP remaining seconds: 20 / 600
Fetch inverter: {{myInverter Serial}}
TX RealTimeRunData Channel: 23 --> 15 82 96 71 68 80 17 07 80 80 0B 00 65 F7 07 70 00 00 00 00 00 00 00 00 29 E0 AF 
Interrupt received
...
WiFi disconnected
Try reconnecting
Network lost connection
WiFi disconnected
Try reconnecting
...
Fetch inverter: {{myInverter Serial}}
Fetch inverter: {{myInverter Serial}}
Fetch inverter: {{myInverter Serial}}
Disable search for AP... WiFi disconnected
Try reconnecting
Network lost connection
done
Fetch inverter: {{myInverter Serial}}
```
Habe schon alle möglichen Varianten mal getestet
* Wifi 2,4GHz Autokanal auf fix 1 gestellt --> Verbindung instabil
* Wifi 5 GHz abgeschaltet --> Verbindung instabil

Hatte bisher das selbe Hardwaresetup seit ca. April 2023 laufen. Es gab da bisher zwei, max drei Wifi Disconnects über die Betriebszeit, die mit Strom ab- und wieder anschalten behoben werden konnten.

**Reply by skippgh** [reply]
Hallo djiwondee
also bei mir läuft es nach wie vor stabil. Und ich habe quasi das gleiche Mesh mit per Kabel verbundenem Repeater 3000, der an einer 7590 FB hängt. Welchen Kanal nimmt denn der Repeater ? 
Nachtrag: meine installierte Version ist v24.1.4 (also nicht mehr ganz aktuell ...)

**Reply by djiwondee** [reply]
Hallo @skippgh,

mittlerweile habe ich bei AVM mal einen Case geöffnet. OpenDTU will sich einfach NUR in das Repeater-WIFI im Keller einbuchen, selbst wenn sich der OpenDTU-ESP32 im selben Raum (Dachgeschoss), wie der Router befindet.

Als Workaround habe ich das OpenDTU Modul in das Erdgeschoß, näher an den Repeater "verbannt". Der Wechselrichter auf dem Dach ist da nach wie vor im Sendebereich. Der Repeater selber nutzt aktuell Kanal 13, wobei mein WLAN Setup aktuell wie folgt konfiguriert ist

<img width="659" alt="image" src="https://github.com/tbnobody/OpenDTU/assets/37173958/b4c40a73-1e51-4a7f-a883-a9d69db794c1">

Ich benutze aktuell OpenDTU [v24.3.15](https://github.com/tbnobody/OpenDTU/releases/tag/v24.3.15). Aber ich glaube, die ESPs haben ein Problem mit dem WIFI in der Router/Repeater-Konstellation. Habe mal einen versuchsweise einen ESP32 und einen ES8266 mit einem simplen Wifiscanner geflashed. Beide wollten sich einfach nicht mit dem Router direkt verbinden...

**Reply by hoeni** [reply]
Nur zur Sicherheit, vielleicht hilft das hier bei der Eingrenzung (und wenn nicht, dann vielleicht jemandem, der wie ich über Google hier landet):

Mein OpenDTU hat seit >1 Jahr ohne Aussetzer funktioniert mit einer Fritzbox 7490 und einer 3000 Repeater. Ich hatte mich nie dafür interessiert, wo sich die DTU verbindet, weil sie genau zwischen den beiden ist.

Heute musste ich aus Gründen die Fritz Box rebooten. Danach konnte ich die OpenDTU nicht mehr erreichen. Auf der seriellen Konsole das übliche WiFi disconnected / Try reconnecting / Network lost connection...

Ich habe alles Mögliche versucht. Komplett zurückgesetzt, neu geflasht (Binary heruntergeladen), selbst compiliert mit PlattformIO, usw. alles ohne Erfolg.

Als ich mich dann mal am Repeater angemeldet habe, habe ich gesehen, **dass der Repeater keine Mesh-Verbindung mehr hatte,** sondern stillschweigend ein WLAN mit Default-Namen geöffnet hatte. Nachdem ich den Repeater mit Konfigübernahme wieder ins Mesh aufgenommen hatte, hat sich die DTU sofort wieder verbunden, seitdem läuft wieder alles problemlos.

**Reply by djiwondee** [reply]
Hallo @hoeni ,
mein Problem hat sich durch den Austausch der FritzBox (7590) erledigt. Die Box ist von 2018 und die 2,4GHz WLAN-Einheit des Gerätes wohl durch ein[ FritzOS-Update teilweise zerstört](https://www.ip-phone-forum.de/threads/fritzbox-7590-2-4ghz-wlan-nur-noch-sehr-sehr-schwach.314938/) worden. Alle hier beschriebenen Symptome sind bei mir auch vorhanden.
Ich habe bei AVM einen Support-Ticket offen. Sie kennen _dieses_ Problem an der 7590 wohl... Für den von AVM angebotenen Austausch ist das Gerät halt leider zu alt.

**Reply by hoeni** [reply]
Hi @djiwondee ,

Mensch, vielen Dank, dass du das geschrieben hast!

Ich habe mal einen Blick in *mein* Mesh geworfen und - siehe da - auch meine Fritz!Box ist im 2.4GHz Band quasi tot, daher konnten sich meine ganzen IoT-Gadgets nur noch am Repeater verbinden. Als der dann weg war, ging dann natürlich gar nichts mehr.
Das erklärt auch so manches seltsame Verhalten, das ich in den letzten Monaten mit anderen Geräten hatte...

Meine Fritz!Box habe ich vom Provider, dort gleich reklamiert und das Tauschgerät ist schon unterwegs.
